ucsdscheduleme/Repo/ScheduleRepo.cs

```diff
@@ -210,5 +210,476 @@ private bool IsWithinTime(List<Section> schedule, int Earliest, int Latest)
+                int[] currDays = { 0, 0, 0, 0, 0 };
+                foreach (Section section in schedule)
+                {
+                    foreach (Meeting meeting in section.Meetings)
+                    {
+                        if (meeting.MeetingType != MeetingType.Final || meeting.MeetingType != MeetingType.Midterm
+                            || meeting.MeetingType != MeetingType.Review)
+                        {
+                            if (meeting.Days.HasFlag(Days.Monday))
+                            {
+                                currDays[0]++;
+                            }
+                            if (meeting.Days.HasFlag(Days.Tuesday))
+                            {
+                                currDays[1]++;
+                            }
+                            if (meeting.Days.HasFlag(Days.Wednesday))
+                            {
+                                currDays[2]++;
+                            }
+                            if (meeting.Days.HasFlag(Days.Thursday))
+                            {
+                                currDays[3]++;
+                            }
+                            if (meeting.Days.HasFlag(Days.Friday))
+                            {
+                                currDays[4]++;
+                            }
+                        }
+                    }
+                }
+
+                // Totals the days scheduled.
+                foreach (int day in currDays)
+                {
+                    if (day != 0)
+                    {
+                        numDays++;
+                    }
+                }
+
+                // Compares the current schedule days to the current max days.
+                if (numDays > mostDay)
+                {
+                    mostDay = numDays;
+                    mostDaySchedule = schedule;
+                }
+            }
+            return mostDaySchedule;
+        } 
+
+        /// <summary>
+        /// Returns the schedule with the least gaps in between classes.
+        /// </summary>
+        /// <returns>The schedule with the least gaps.</returns>
+        /// <param name="possibleSchedules">Possible schedules with no time conflicts.</param>
+        public List<Section> LeastGaps(List<List<Section>> possibleSchedules)
+        {
+            List<Section> leastGapsSchedule = possibleSchedules[0];
+            int leastGap = 99999;
+
+            // Iterate through the possible schedules and separate the times into respective days.
+            foreach (List<Section> schedule in possibleSchedules)
+            {
+                List<List<Tuple<int, int>>> week = new List<List<Tuple<int, int>>>();
+
+                foreach (Section section in schedule)
+                {
+                    foreach (Meeting meeting in section.Meetings)
+                    {
+                        if (meeting.Days.HasFlag(Days.Monday))
+                        {
```

Comment:
Meeting was of different types, so was counting for the meeting type of final and reviews as well.